browser/mapping.go

```diff
@@ -339,15 +339,22 @@ func mapFrame(vu moduleVU, f *common.Frame) mapping {
 			return mapElementHandle(vu, fe), nil
 		},
 		"getAttribute": f.GetAttribute,
-		"goto": func(url string, opts goja.Value) *goja.Promise {
+		"goto": func(url string, opts goja.Value) (*goja.Promise, error) {
+			gopts := common.NewFrameGotoOptions(
```

Comment:
Should we rename `FrameGotoOptions` to `GotoOptions` as both `Frame` and `Page` share the same type? And saying `FrameGotoOptions` in the `Page` signature feels a bit off.

Reply:
I'm fine with having `FrameGotoOptions` in `page.goto` since `page.goto` uses `frame`'s implementation. If we refactored it out of `frame`, then I'd go down the route of renaming `FrameGotoOptions`.

Reply:
If the `goto` implementation was moved out of `frame` and into a new file (e.g. `navigation`) then i think it would make sense to rename the the option. I think that the scope of that work is outside of this PR though.

Reply:
Thanks for the clarification 👍